test/OpenTelemetry.Instrumentation.AWSLambda.Tests/Implementation/AWSMessagingUtilsTests.cs

```diff
@@ -0,0 +1,79 @@
+// <copyright file="AWSMessagingUtilsTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Amazon.Lambda.SQSEvents;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Instrumentation.AWSLambda.Implementation;
+using OpenTelemetry.Trace;
+using Xunit;
+using static Amazon.Lambda.SQSEvents.SQSEvent;
+
+namespace OpenTelemetry.Instrumentation.AWSLambda.Tests.Implementation;
+
+public class AWSMessagingUtilsTests
+{
+    private const string TraceId = "0af7651916cd43dd8448eb211c80319c";
+    private const string SpanId1 = "b9c7c989f97918e1";
+    private const string SpanId2 = "b9c7c989f97918e2";
+
+    public AWSMessagingUtilsTests()
+    {
+        var tracerProvider = Sdk.CreateTracerProviderBuilder()
+            .Build();
+    }
+
+    [Fact]
+    public void ExtractParentContext_SetParentFromMessageBatchIsDisabled_ParentIsNotSet()
+    {
+        AWSMessagingUtils.SetParentFromMessageBatch = false;
+        var @event = CreateSqsEventWithMessages(new[] { SpanId1, SpanId2 });
```

Comment:
Maybe use "input" to avoid the weird @-syntax

Reply:
Renamed to sqsEvent.